src/types/index.d.ts

```diff
@@ -208,7 +208,7 @@ export type DataPlaneNetworking = {
   }[]
   gateway?: {
     tags: Record<string, string>
-    type?: 'builtin' | 'delegated' | undefined
+    type?: 'BUILTIN'
```

Comment:
I think technically 'DELEGATED' is a type that exists

Reply:
I’ve updated this and added a comment:

```ts
/**
 * The presence of the `gateway` field means one of two things:
 *
 * - The resource represents a builtin gateway (when `gateway.type === 'BUILTIN'`)
 * - The resource represents a delegated gateway (when `gateway.type === 'DELEGATED'` or `gateway.type` is omitted)
 *
 * If the `gateway` field is absent, the resource represents a regular Data Plane Proxy.
 */
gateway?: {
  tags: Record<string, string>
  type?: 'BUILTIN' | 'DELEGATED'
}
```